settings.php

```diff
@@ -1047,6 +1047,22 @@ function convertseconds($argument)
                                     </div>
                                 </div>
                             </form>
+                            <div class="modal fade" id="apiTokenModal" role="dialog" data-keyboard="false"
+                                tabindex="-1" data-backdrop="static" aria-labelledby="apiTokenModal">
+                                <div class="modal-dialog" role="document">
+                                    <div class="modal-content">
+                                        <div class="modal-header">
+                                            <h4 class="modal-title" id="apiTokenModalHeaderLabel">API Token</h4>
+                                        </div>
+                                        <div class="modal-body">
+                                        <pre><iframe id="apiTokenIframe" name="apiToken_iframe" src="scripts/pi-hole/php/api_token.php"></iframe></pre>
```

Comment:
The QR code doesn't look great on mobile. Because it's built using a table the height can only be responsive to a certain extent. I tried using the `createImage()` function from the QR code lib being used but no dice.

Not sure how important this is since if you're accessing the API token on mobile you most likely will be copying the raw string rather than using another mobile device to take a photo of the qr code.

Reply:
In my opinion it's an edge case. Generating a qr code with a mobile to be scanned with another mobile.

Reply:
Agreed. This is how it looks on my screen:
![Screenshot_20211209-205750.png](https://user-images.githubusercontent.com/10479093/145504535-5a70f406-c035-4d96-b54e-29eba16a03f2.png)

Reply:
Looks good. Could you add support for the theme's font color as we did in https://github.com/pi-hole/AdminLTE/pull/1934. (Otherwise it's black font on black background in the dark themes)

Reply:
I think it might be safer to always use white background and black text for the `<pre>` element that displays the QRcode (even in dark themes).

Reply:
Regarding the distorted QRcode, I think most readers are able to ignore these little distortions.

Reply:
> I think it might be safer 

What do you mean with safer in this regard? Safer to be successfully scanned?

Reply:
Yes, that's what a mean.
To be correctly scanned, the QR code should have a margin (https://www.qrcode.com/en/howto/code.html).

This code creates a image without a margin, but using a light background (white or light gray) has enough contrast to be consider a margin.
On the other hand, with a dark background there will be no discernible margin. 
I'm not sure, but I think some scanners may have difficulty reading the image without margin.

This solution should be tested with dark themes.

Reply:
You're right! I tested the PR it with deep dark theme and my mobile could **not** scan the code. After switching to the dark or light theme I had no issues to scan the code.

Reply:
I think the problem is the low contrast between the background and the edges of the QR code.

My sugestion:
Adapt the code from #1934 to use white as the QR code's background color for all themes.

![qrcode-margin_suggestion](https://user-images.githubusercontent.com/1385443/145721284-7978bb8f-4cdd-458e-bbd0-b581a8efd954.gif)

Reply:
Updated to use a white background no matter the theme. My QR scanner worked with each theme. Thanks for pointing this out and brainstorming on this!

Reply:
Just a small nit: could could you remove (or color white) the small leftover margin. Best seen in LCARS theme

![Bildschirmfoto zu 2021-12-14 07-11-32](https://user-images.githubusercontent.com/26622301/145943298-5964327d-9058-4f44-9a59-99735f345307.png)

Reply:
It's the `<pre>` background-color. 

You just need to add this into _lcars.css_:

```css
#apiTokenModal pre {
    background: #FFF;
}
```

Reply:
It's also in the other themes!

Reply:
Edited.